bgpd/bgp_attr.c

```diff
@@ -3635,10 +3636,22 @@ size_t bgp_packet_mpattr_start(struct stream *s, struct peer *peer, afi_t afi,
 			stream_put_ipv4(s, attr->nexthop.s_addr);
 			break;
 		case SAFI_MPLS_VPN:
-			stream_putc(s, 12);
-			stream_putl(s, 0); /* RD = 0, per RFC */
-			stream_putl(s, 0);
-			stream_put(s, &attr->mp_nexthop_global_in, 4);
+			plk = peer_lookup_by_host(NULL, peer->host);
```

Comment:
What is the point of this function? You are looking for a peer while you have a peer?

Reply:
Print some debug info after line 3639:

zlog_debug("%s (%d): peer=%p(%s-%s), plk=%p(%s-%s)", __func__, __LINE__, peer, peer->hostname, peer->host, plk, plk->hostname, plk->host);

root@debian:/# grep -n "bgp_packet_mpattr_start" tmp/frr.log
8163:2022/06/09 06:19:21 BGP: [H5JS8-HGGFQ] bgp_packet_mpattr_start (3640): peer=0x560d6d15b5f0((null)-fcff:0:1::1), plk=0x7ff05a8a2010(debian-fcff:0:1::1)
8453:2022/06/09 06:19:21 BGP: [H5JS8-HGGFQ] bgp_packet_mpattr_start (3640): peer=0x560d6d15b5f0((null)-fcff:0:1::1), plk=0x7ff05a8a2010(debian-fcff:0:1::1)

So, here we have two different peers.

Reply:
About your question "Do we need an additional topotest covering these fixes?" - we plan to continue with RouteReflector functionality, so possibly after it.

Reply:
The existing topo test was already aligned to updated label calculation algorythm. All other changes concern the correct values in BGP update fields. Could you kindly advise which test to use for BGP update content?

Reply:
The idea of peer_lookup_by_hostname() is to find a peer structure by string, but here (seems) we do something wrong.

Reply:
Fix in #11522